### ISSUE TITLE
RefinementList searchForItems

### ISSUE THREAD

**Issue by iam4x** [algolia/instantsearch.js#2552]
The connector `connectRefinementList` output two different structure of item, the label is missing when using the searchForItems:

* without search:
![image](https://user-images.githubusercontent.com/893837/32441371-9a120406-c2f7-11e7-8879-baf1adf132c4.png)

* with search:
![image](https://user-images.githubusercontent.com/893837/32441386-a881105e-c2f7-11e7-917e-3c5e896a71da.png)

----

Another issue, in the dev-novel example https://community.algolia.com/instantsearch.js/v2/dev-novel/?selectedStory=RefinementList.with%20search%20inside%20items the searchbar is always disabled.

```jsx
const shouldDisableSearchInput =
      this.props.searchIsAlwaysActive !== true &&
      !(this.props.isFromSearch || !this.props.hasExhaustiveItems);
```

It comes from the `this.props.hasExhaustiveItems` which I changed recently.

**Comment by bobylito** [comment]
I found the second part of the bug on friday. I'm almost done with the fix

**Comment by bobylito** [comment]
Also related to https://github.com/algolia/instantsearch.js/issues/2454

**Comment by bobylito** [comment]
This is fixed in 2.2.3
